### PR TITLE
[Keyboard] add MT64RGB

### PR DESCRIPTION
**Pull request by 704340378**
## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mt64rgb/config.h [hunk 1]
```diff
@@ -0,0 +1,87 @@
+#pragma once
```
Comment (drashna) [inline keyboards/mt64rgb/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/mt64rgb/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/mt64rgb/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/mt64rgb/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#include "i2c_master.h"
+
+extern rgb_config_t rgb_matrix_config;
+#endif
+
+// extern keymap_config_t keymap_config;
+
+#ifdef RGBLIGHT_ENABLE
+extern rgblight_config_t rgblight_config;
+#endif
```
Comment (drashna) [inline keyboards/mt64rgb/keymaps/default/keymap.c:14]: This entire block isn't needed and shouldn't be added here. 

```suggestion
```

File: keyboards/mt64rgb/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,44 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/mt64rgb/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/mt64rgb/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+	    switch (biton32(layer_state)) {
+	      case 1:
+	        rgb_matrix_set_color(77,0xFF, 0x80, 0x00);
+          break;
+      }
+    }
+    if ( this_led & (1<<USB_LED_CAPS_LOCK)) {
+          rgb_matrix_set_color(28, 0xFF, 0xFF, 0xFF);
+    }
+}
```
Comment (drashna) [inline keyboards/mt64rgb/keymaps/via/keymap.c:44]:
```suggestion
void rgb_matrix_indicators_user(void) {
    if (!g_suspend_state && layer_state_is(1)) {
        rgb_matrix_set_color(77,0xFF, 0x80, 0x00);
    }
    if (host_keyboard_led_state().caps_lock) {
          rgb_matrix_set_color(28, 0xFF, 0xFF, 0xFF);
    }
}
```

File: keyboards/mt64rgb/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	    switch (biton32(layer_state)) {
+	      case 1:
+	        rgb_matrix_set_color(77,0xFF, 0x80, 0x00);
+          break;
+      }
+    }
+    if ( this_led & (1<<USB_LED_CAPS_LOCK)) {
+          rgb_matrix_set_color(28, 0xFF, 0xFF, 0xFF);
+    }
+}
```
Comment (drashna) [inline keyboards/mt64rgb/keymaps/default/keymap.c:48]: This entire block could be condensed into: 
```suggestion
void rgb_matrix_indicators_user(void) {
    if (!g_suspend_state && layer_state_is(1)) {
        rgb_matrix_set_color(77,0xFF, 0x80, 0x00);
    }
    if (host_keyboard_led_state().caps_lock) {
          rgb_matrix_set_color(28, 0xFF, 0xFF, 0xFF);
    }
}
```

File: keyboards/mt64rgb/mt64rgb.c [hunk 7]
```diff
@@ -0,0 +1,101 @@
+#include "mt64rgb.h"
```
Comment (drashna) [inline keyboards/mt64rgb/mt64rgb.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/mt64rgb/mt64rgb.h [hunk 8]
```diff
@@ -0,0 +1,18 @@
+#pragma once
```
Comment (drashna) [inline keyboards/mt64rgb/mt64rgb.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mt64rgb/rules.mk [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/mt64rgb/rules.mk:11]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mt64rgb/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+#ifdef RGBLIGHT_ENABLE
+extern rgblight_config_t rgblight_config;
+#endif
+
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    [0] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/mt64rgb/keymaps/default/keymap.c:20]:
```suggestion
    [0] = LAYOUT_64_ansi(
```

File: keyboards/mt64rgb/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,  KC_UP,  KC_DEL,
+        KC_LCTL,  KC_LALT, KC_LWIN,                     KC_SPC,                          MO(1),  KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+                        ),
+    [1]   = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/mt64rgb/keymaps/default/keymap.c:27]:
```suggestion
    [1]   = LAYOUT_64_ansi(
```

File: keyboards/mt64rgb/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,44 @@
+#include QMK_KEYBOARD_H
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	[0] =LAYOUT(
```
Comment (noroadsleft) [inline keyboards/mt64rgb/keymaps/via/keymap.c:6]:
```suggestion
	[0] =LAYOUT_64_ansi(
```

File: keyboards/mt64rgb/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,  KC_UP,  KC_DEL,
+        KC_LCTL,    KC_LALT, KC_LWIN,              KC_SPC,                                          MO(1),                   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+                        ),
+	[1]   =LAYOUT(
```
Comment (noroadsleft) [inline keyboards/mt64rgb/keymaps/via/keymap.c:13]:
```suggestion
	[1]   =LAYOUT_64_ansi(
```

File: keyboards/mt64rgb/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+		KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,
+		KC_TRNS, KC_TRNS, KC_UP, KC_TRNS, KC_TRNS, RGB_SAI, RGB_SAD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET,
+		KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  BL_TOGG, BL_INC,  BL_STEP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,RGB_VAI, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, RGB_TOG, KC_TRNS, RGB_MOD, RGB_HUI, RGB_VAD, RGB_HUD),
+	LAYOUT(
```
Comment (noroadsleft) [inline keyboards/mt64rgb/keymaps/via/keymap.c:19]:
```suggestion
	[2] = LAYOUT_64_ansi(
```

File: keyboards/mt64rgb/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, BL_DEC,  BL_TOGG, BL_INC,  BL_STEP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+	LAYOUT(
```
Comment (noroadsleft) [inline keyboards/mt64rgb/keymaps/via/keymap.c:25]:
```suggestion
	[3] = LAYOUT_64_ansi(
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/10865/files).

File: keyboards/mt64rgb/rules.mk (hunk 9 above)
Comment (704340378) [reply keyboards/mt64rgb/rules.mk:11]: Sorry i dont find different in this conversation

**Comment by drashna** [comment]
Don't forget about the hidden suggestions. 

Also, if you head to the "Files changed" tab, you can queue a bunch of changes and apply them all at once.

File: keyboards/mt64rgb/config.h [hunk 16]
```diff
@@ -0,0 +1,99 @@
+* Copyright 2020 MT
```
Comment (fauxpark) [inline keyboards/mt64rgb/config.h:1]:
```suggestion
/* Copyright 2020 MT 
```

File: keyboards/mt64rgb/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,46 @@
+* Copyright 2020 MT
```
Comment (fauxpark) [inline keyboards/mt64rgb/keymaps/default/keymap.c:1]:
```suggestion
/* Copyright 2020 MT 
```

File: keyboards/mt64rgb/readme.md [hunk 18]
```diff
@@ -0,0 +1,16 @@
+# mt64rgb 
+
+![mt64rgb](https://i.imgur.com/kxGGUg9.jpg?1)
+
+A 60% keyboard PCB made by MT.
+It supports standard 64 keyboard，with hotswap and RGB backlight.
+
+ Keyboard Maintainer: [MT](https://github.com/704340378)
```
Comment (fauxpark) [inline keyboards/mt64rgb/readme.md:8]:
```suggestion
* Keyboard Maintainer: [MT](https://github.com/704340378)

```
Please also change these files to Unix line endings.

File: keyboards/mt64rgb/rules.mk [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+RGB_MATRIX_ENABLE = IS31FL3733 # Use RGB matrix
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
```
Comment (fauxpark) [inline keyboards/mt64rgb/rules.mk:22]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mt64rgb/config.h [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4D54  // "MT"
+#define PRODUCT_ID      0x0001
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    mt
+#define PRODUCT         mt64rgb
+#define DESCRIPTION     mt64rgb
```
[the remark this answers is not in the document]
Comment (704340378) [reply keyboards/mt64rgb/config.h]: I DELETE this line ,but i dont know right or no

File: keyboards/mt64rgb/config.h [hunk 21]
```diff
@@ -1,14 +1,25 @@
...
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  */ 
 #pragma once
 
 #include "config_common.h"
 
 /* USB Device descriptor parameter */
 #define VENDOR_ID       0x4D54  // "MT"
 #define PRODUCT_ID      0x0001
```
Comment (noroadsleft) [inline keyboards/mt64rgb/config.h:22]:
```suggestion
#define PRODUCT_ID      0x0001
#define DEVICE_VER      0x0001
#define MANUFACTURER    MT
#define PRODUCT         MT64RGB
```

File: keyboards/mt64rgb/readme.md [hunk 22]
```diff
@@ -1,17 +1,16 @@
 # mt64rgb 
 
-![mt64rgb](https://imgur.com/z5fxpTO)
+![mt64rgb](https://i.imgur.com/kxGGUg9.jpg?1)
 
 A 60% keyboard PCB made by MT.
-It supports standard 64 keyboard，with hotswap and rgbbacklight
- GH60 layout and mutultiple layouts like arrows, 2u shift, minila, etc.
-Fitted with all stock GH60 cases, with WS2812 RGB underglow.
+It supports standard 64 keyboard，with hotswap and RGB backlight.
```
Comment (noroadsleft) [inline keyboards/mt64rgb/readme.md:6]:
```suggestion
It supports an ANSI-based 64-key layout, with hotswap sockets and per-key RGB backlight.
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/mt64rgb/rules.mk [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+MOUSEKEY_ENABLE = no      # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```
Comment (noroadsleft) [inline keyboards/mt64rgb/rules.mk:20]: I don't actually see any underglow LEDs in the photo that's on the readme. Looks like all the LEDs face the top side of the PCB. Is that correct?

Comment (704340378) [reply keyboards/mt64rgb/rules.mk:20]: sorry.i will add underglow led in the next version,so i think maybe we can still keep this code .thank you .
https://i.imgur.com/pdjRUwg.png

File: keyboards/mt64rgb/config.h (hunk 21 above)
Comment (704340378) [reply keyboards/mt64rgb/config.h:22]: thank you very much

File: keyboards/mt64rgb/readme.md (hunk 22 above)
Comment (704340378) [reply keyboards/mt64rgb/readme.md]: will do ,thank you

File: keyboards/mt64rgb/rules.mk [hunk 24]
```diff
@@ -0,0 +1,25 @@
...
+RGB_MATRIX_ENABLE = IS31FL3733 # Use RGB matrix
```
Comment (noroadsleft) [inline keyboards/mt64rgb/rules.mk:21]: `RGB_MATRIX_ENABLE` has been split up into two definitions. `RGB_MATRIX_ENABLE` is now a `yes` or `no`, and `RGB_MATRIX_DRIVER` should be set to the driver being used.


```suggestion
RGB_MATRIX_ENABLE = yes     # Use RGB matrix
RGB_MATRIX_DRIVER = IS31FL3733
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

File: keyboards/mt64rgb/readme.md [hunk 25]
```diff
@@ -0,0 +1,15 @@
+# mt64rgb 
+
+![mt64rgb](https://i.imgur.com/kxGGUg9.jpg?1)
+
+A 60% keyboard PCB made by MT.
+It supports an ANSI-based 64-key layout, with hotswap sockets and per-key RGB backlight.
+* Keyboard Maintainer: [MT](https://github.com/704340378)
```
Comment (fauxpark) [inline keyboards/mt64rgb/readme.md:7]:
```suggestion

* Keyboard Maintainer: [MT](https://github.com/704340378)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mt64rgb/config.h [hunk 26]
```diff
@@ -0,0 +1,102 @@
...
+ *
+ */
+#define MATRIX_ROW_PINS { D7, D6, D5, D3, D2 }
+#define MATRIX_COL_PINS {B5, B6, C6, C7, F7, F6,F5, F4, F1, F0, B1, B2, B3, B7 }
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+
+#ifdef RGB_MATRIX_ENABLE
```
Comment (drashna) [inline keyboards/mt64rgb/config.h:49]: Agh, the inconsistent indentation ...

you should be able to run `qmk cformat` on the folder, or clang-format, manually.  This should help reformat the files so they're consistent, and fit with our coding conventions.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**